packages/core-mobile/app/new/contexts/SignupProvider.tsx

```diff
@@ -0,0 +1,117 @@
+import { TotpChallenge } from '@cubist-labs/cubesigner-sdk'
+import React, {
+  createContext,
+  Dispatch,
+  ReactNode,
+  SetStateAction,
+  useCallback,
+  useContext,
+  useMemo,
+  useState
+} from 'react'
+import SeedlessService from 'seedless/services/SeedlessService'
+import AnalyticsService from 'services/analytics/AnalyticsService'
+import { useRouter } from 'expo-router'
+import { copyToClipboard } from 'new/utils/clipboard'
+import { TotpErrors } from 'seedless/errors'
+import { Result } from 'types/result'
+import { hideLogoModal, showLogoModal } from 'new/components/LogoModal'
+import { OidcAuth } from 'new/types'
+
+export interface SignupContextState {
+  handleAccountVerified: () => Promise<void>
+  handleCopyCode: () => void
+  onVerifyCode: (code: string) => Promise<Result<undefined, TotpErrors>>
+  onVerifySuccess: () => void
+  totpKey?: string
+  totpChallenge?: TotpChallenge
+  setTotpChallenge: Dispatch<SetStateAction<TotpChallenge | undefined>>
+  oidcAuth?: OidcAuth
+  setOidcAuth: Dispatch<SetStateAction<OidcAuth | undefined>>
+  allowsUserToAddLater: boolean
```

Comment:
When navigating with expo-router, we can’t pass callbacks, but we can pass variables like booleans. So I think we don’t need to manage `allowsUserToAddLater` with context.

Reply:
<img width="447" alt="Screenshot 2024-12-18 at 3 41 32 PM" src="https://github.com/user-attachments/assets/b8528d60-7521-44b8-b71b-1e4193de1523" />

I think searchParams only allows string or string[]

Reply:
You’re right. However, wouldn’t it be better to pass it as a parameter, even after converting the string to a boolean?

Reply:
moved `allowsUserToAddLater` to search params.